In [1]:
import pandas as pd
from scipy.stats import chi2_contingency, chi

In [4]:
campagin_data = pd.read_excel('grocery_database.xlsx', sheet_name='campaign_data')

In [5]:
campagin_data.head()

customer_id  campaign_name campaign_date mailer_type  signup_flag
0           74  delivery_club    2020-07-01     Mailer1            1
1          524  delivery_club    2020-07-01     Mailer1            1
2          607  delivery_club    2020-07-01     Mailer2            1
3          343  delivery_club    2020-07-01     Mailer1            0
4          322  delivery_club    2020-07-01     Mailer2            1

In [7]:
campagin_data['mailer_type'].unique()

array(['Mailer1', 'Mailer2', 'Control'], dtype=object)

In [10]:
# I am only interested in Mailter type not equl to control
new_campagin_data = campagin_data.loc[campagin_data['mailer_type'] != 'Control']

In [11]:
new_campagin_data.head(10)

customer_id  campaign_name campaign_date mailer_type  signup_flag
0           74  delivery_club    2020-07-01     Mailer1            1
1          524  delivery_club    2020-07-01     Mailer1            1
2          607  delivery_club    2020-07-01     Mailer2            1
3          343  delivery_club    2020-07-01     Mailer1            0
4          322  delivery_club    2020-07-01     Mailer2            1
5          115  delivery_club    2020-07-01     Mailer2            0
6            1  delivery_club    2020-07-01     Mailer2            1
7          120  delivery_club    2020-07-01     Mailer1            1
8           52  delivery_club    2020-07-01     Mailer1            1
9          405  delivery_club    2020-07-01     Mailer1            0

In [12]:
print(campagin_data.shape, new_campagin_data.shape)

(870, 5) (711, 5)


In [14]:
# Let's observe the values
observed_values = pd.crosstab(new_campagin_data['mailer_type'], new_campagin_data['signup_flag'])

In [15]:
observed_values

signup_flag    0    1
mailer_type          
Mailer1      252  123
Mailer2      209  127

In [17]:
mailer_1 = 123 / (123 + 252)
mailer_2 = 127 / (127 + 209)

print('The percentag rate for mailer 1 is: {} and the percentage rate for mailer 2 is: {}'.format(mailer_1 , mailer_2 ))

The percentag rate for mailer 1 is: 0.328 and the percentage rate for mailer 2 is: 0.37797619047619047


## The percentage rate for Mailer Types
    1. Mailer - 1 = 32.8%
    2. Mailer - 2 = 37.7%

In [18]:
null_hypothesis = 'There is no relationship between mailer and signup rate. They are independent'
alternate_hypothesis = 'There is a relationship between mailer and signup rate. They are not independent'
acceptance_criteria = 0.05

In [19]:
chi2_statistics, p_value, dof, expected_values = chi2_contingency(observed_values, correction=False) # here the dof is 1, because it's a 2x2 matrix. For all DOF = 1, then correlation = False

In [20]:
print(chi2_statistics, p_value)

1.9414468614812481 0.16351152223398197


In [21]:
# find the critical values
critical_values = chi.ppf(1 - acceptance_criteria, dof)

In [22]:
print(critical_values)

1.9599639845400538


In [23]:
if chi2_statistics >= critical_values:
    print(f'As our chi-square statistcs of {chi2_statistics} is higher than our critical value of {critical_values} - we reject the null hypothesis, and conclude that: {alternate_hypothesis}')
else:
    print(f'As our chi-square statistics of {chi2_statistics} is lower than the critical vlaue of {critical_values} - we retain the null hypothesis, and conclude that {null_hypothesis}')

As our chi-square statistics of 1.9414468614812481 is lower than the critical vlaue of 1.9599639845400538 - we retain the null hypothesis, and conclude that There is no relationship between mailer and signup rate. They are independent


In [24]:
# talking interms of p-value
if p_value <= acceptance_criteria:
    print(f'As our p-value of {p_value} is lower than the acceptance of {acceptance_criteria}. We reject the null hypothesis and we conclude that: {alternate_hypothesis}')
else:
    print(f'As our p-value of {p_value} is higher than the acceptance criteria of {acceptance_criteria}. we retain the null hypothesis and conclude that: {null_hypothesis}')

As our p-value of 0.16351152223398197 is higher than the acceptance criteria of 0.05. we retain the null hypothesis and conclude that: There is no relationship between mailer and signup rate. They are independent
